source/MaterialXGenOsl/OslSyntax.cpp

```diff
@@ -274,7 +317,7 @@ OslSyntax::OslSyntax()
         "emission", "background", "diffuse", "oren_nayer", "translucent", "phong", "ward", "microfacet",
         "reflection", "transparent", "debug", "holdout", "subsurface", 
         // TODO: Add all OSL standard library functions names
-        "mix", "rotate"
+        "mix", "rotate", "textureresource"
```

Comment:
This reminds me, we should add all our custom data type names here. For another PR.

Reply:
I'll log something for this.

Reply:
https://github.com/materialx/MaterialX/issues/781